### PR TITLE
Spider 0.8.2 l3 offload

### PR DESCRIPTION
**Pull request by LKomaryanskiy**

### CODE REVIEW

File: meta-rcar-l3-offload/recipes-kernel/linux-bsp/linux-renesas_5.10.bbappend [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DESCRIPTION = "Linux kernel for the R-Car Gateway based board"
```
Comment (lorc) [inline meta-rcar-l3-offload/recipes-kernel/linux-bsp/linux-renesas_5.10.bbappend:1]: Is not needed

File: meta-rcar-l3-offload/recipes-kernel/linux-bsp/linux-renesas_5.10.bbappend [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+RENESAS_BSP_URL = " \
+    git://github.com/LKomaryanskiy/linux-bsp.git"
```
Comment (lorc) [inline meta-rcar-l3-offload/recipes-kernel/linux-bsp/linux-renesas_5.10.bbappend:4]: Please use XT repo

File: prod-devel-rcar-s4.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+images:
+  full:
+    type: gpt
+    desc: "Full SD-card/eMMC image"
+    partitions:
+      domd_rootfs:
+        gpt_type: B921B045-1DF0-41C3-AF44-4C6F280D3FAE # Linux aarch64 root
+        type: raw_image
+        image_path: "%{YOCTOS_WORK_DIR}/build/tmp/deploy/images/%{MACHINE}/rcar-image-minimal-spider.ext4"
```
Comment (lorc) [inline prod-devel-rcar-s4.yaml:55]: Missing end of line

File: README.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+# dd if=/full.img of=/dev/mmcblk0 bs=1M
+```
+
+For more information about `rouge` check its
+[manual](https://moulin.readthedocs.io/en/latest/rouge.html).
+
+## U-boot environment
+
+It is possible to run the build from TFTP+NFS or eMMC. With NFS boot
+it is possible to configure board IP, server IP and NFS path for DomD.
```
Comment (lorc) [inline README.md:110]: We have no DomD there. Call it just "system"

File: README.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+On main host, connected to TSN0:
+```
+sudo ip route add 192.168.5.0/24 via 192.168.1.1
+```
+
+On host connected to TSN2:
+```
+sudo ifconfig enp1s0 192.168.3.100
+sudo ifconfig virt0 hw ether C8:D7:4A:4E:47:50
```
Comment (lorc) [inline README.md:163]: Why do you need to hard-code MAC address?

File: meta-rcar-l3-offload/recipes-kernel/linux-bsp/linux-renesas_5.10.bbappend (hunk 1 above)
Comment (LKomaryanskiy) [reply meta-rcar-l3-offload/recipes-kernel/linux-bsp/linux-renesas_5.10.bbappend]: Removed

File: meta-rcar-l3-offload/conf/layer.conf [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+# We have recipes-* directories, add to BBFILES
+BBFILES += "${LAYERDIR}/recipes-*/*/*.bb \
+	${LAYERDIR}/recipes-*/*/*.bbappend"
+
+BBFILE_COLLECTIONS += "rswitch"
+BBFILE_PATTERN_rswitch = "^${LAYERDIR}/"
+BBFILE_PRIORITY_rswitch = "6"
+
+LAYERSERIES_COMPAT_rswitch = "dunfell"
```
[the remark this answers is not in the document]
Comment (LKomaryanskiy) [reply meta-rcar-l3-offload/conf/layer.conf]: Added end of line

File: meta-rcar-l3-offload/recipes-kernel/linux-bsp/linux-renesas_5.10.bbappend (hunk 2 above)
Comment (LKomaryanskiy) [reply meta-rcar-l3-offload/recipes-kernel/linux-bsp/linux-renesas_5.10.bbappend]: Repo has been changed

File: prod-devel-rcar-s4.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
+desc: "R-Switch L3 HW offload development setup for Renesas RCAR Gen4 hardware"
+min_ver: "0.4"
+
+variables:
+  YOCTOS_WORK_DIR: "yocto"
+  MACHINE: "spider"
+  SOC_FAMILY: "r8a779f0"
+
+components:
+  domd:
```
[the remark this answers is not in the document]
Comment (LKomaryanskiy) [reply prod-devel-rcar-s4.yaml]: Renamed

File: prod-devel-rcar-s4.yaml (hunk 3 above)
Comment (LKomaryanskiy) [reply prod-devel-rcar-s4.yaml]: Added end of line

File: README.md (hunk 4 above)
Comment (LKomaryanskiy) [reply README.md]: Renamed. Also, renamed DomD naming in another place

File: README.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+sudo ip link add virt0 type dummy
+sudo ifconfig virt0 hw ether C8:D7:4A:4E:47:50
+sudo ifconfig virt0 up 192.168.5.100
+```
+
+Ping the virt interface, from host or namespace connected to TSN0:
```
[the remark this answers is not in the document]
Comment (LKomaryanskiy) [reply README.md]: The test setup has been simplified to forward messages from one interface to another.

File: README.md (hunk 5 above)
Comment (LKomaryanskiy) [reply README.md]: It was just an example, you can use any other valid MAC address. As the test setup was simplified, it was removed too.

**Pull request merged**